src/main/java/uk/gov/hmcts/cmc/claimstore/services/CountyCourtJudgmentService.java

```diff
@@ -69,15 +71,28 @@ public Claim save(final long submitterId, final CountyCourtJudgment countyCourtJ
             );
         }
 
+        if (isCountyCourtJudgmentInputValid(countyCourtJudgment)) {
+            throw new UnprocessableEntityException("Given data input is invalid");
+        }
+
         claimRepository.saveCountyCourtJudgment(claimId, jsonMapper.toJson(countyCourtJudgment));
 
         Claim claimWithCCJ = getClaim(claimId);
 
-        eventProducer.createCountyCourtJudgmentSubmittedEvent(claimWithCCJ);
+        eventProducer.createCountyCourtJudgmentRequestedEvent(claimWithCCJ);
 
         return claimWithCCJ;
     }
 
+    private boolean isCountyCourtJudgmentInputValid(CountyCourtJudgment ccj) {
+        return ((ccj.getPaymentOption().equals(PaymentOption.IMMEDIATELY)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS1067)